### PR TITLE
reclean two EBs to fix divergence

### PR DESCRIPTION
**Pull request by pyhsiehATalma**
Reclean Sgr_A_st_ao_03_TM1 (spw 35) and Sgr_A_st_t_03_7M (spw 24) to fix the divergence in pipeline product.

### CODE REVIEW

File: pipeline_scripts/override_tclean_commands.json [hunk 1]
```diff
@@ -22,4 +22,36 @@
...
+        "lownoisethreshold": 2.0,
+        "growiterations": 75
+        "sidelobethreshold": 1.25,
+        "negativethreshold": 0.0,
+        "smoothfactor": 0.5,
+        "minbeamfrac": 0.1,
+        "cutthreshold": 0.01
+      },
+    }
+  }
```
Comment (keflavich) [inline pipeline_scripts/override_tclean_commands.json:40]:
```suggestion
  },
```

File: pipeline_scripts/override_tclean_commands.json [hunk 2]
```diff
@@ -22,4 +22,36 @@
       }
     }
   }
```
Comment (keflavich) [inline pipeline_scripts/override_tclean_commands.json:24]:
```suggestion
  },
```

**Review by keflavich — CHANGES_REQUESTED** [review]
@pyhsiehATalma could you validate the file?  I think you need commas.

You can do:
```bash
jq . override_tclean_commands.json
```

or
```python
import json, pprint
with open('override_tclean_commands.json', 'r') as fh:
    cmds = json.load(fh)
pprint.pprint(cmds)
```

**Comment by keflavich** [comment]
@pyhsiehATalma are you going to work on this further?  Or were these changes already incorporated into another PR?

**Comment by pyhsiehATalma** [comment]
{
  "Sgr_A_st_c_03_7M": {
    "tclean_cont_pars": {
      "aggregate": {
        "imagename": "uid___A001_X15a0_Xae.s36_0.Sgr_A_star_sci.spw16_18_20_22_24_26.cont.I.iter1.reclean",
        "threshold": "0.0024Jy",
        "noisethreshold": 3.0,
        "lownoisethreshold": 1.75,
        "growiterations": 150
      }
    }
  },
  "Sgr_A_st_ar_03_7M": {
    "tclean_cube_pars": {
      "spw24": {
        "imagename": "uid___A001_X15a0_X1a4.s38_0.Sgr_A_star_sci.spw24.cube.I.iter1.reclean",
        "cyclefactor": 2.0
      },
      "spw26": {
        "imagename": "uid___A001_X15a0_X1a4.s38_0.Sgr_A_star_sci.spw26.cube.I.iter1.reclean",
        "cyclefactor": 2.0
      }
    }
  },
  "Sgr_A_st_ao_03_TM1": {
    "tclean_cube_pars": {
      "spw35": {
        "imagename": "uid___A001_X15a0_X190.s38_0.Sgr_A_star_sci.spw35.cube.I.iter1.reclean",
	"threshold": "0.01Jy",
	"noisethreshold": 3.25,
	"lownoisethreshold": 1.5,
	"growiterations": 75,
	"sidelobethreshold": 2.0,
	"negativethreshold": 10.0,
	"smoothfactor": 1.0,
	"minbeamfrac": 0.3,
	"cutthreshold": 0.01
      }
    }
  },
  "Sgr_A_st_ao_03_TM1": {
    "tclean_cube_pars": {
      "spw35": {
        "imagename": "uid___A001_X15a0_X190.s38_0.Sgr_A_star_sci.spw33.cube.I.iter1.reclean",
	"threshold": "0.013Jy",
	"noisethreshold": 3.25,
	"lownoisethreshold": 1.5,
	"growiterations": 75,
	"sidelobethreshold": 2.0,
	"negativethreshold": 15.0,
	"smoothfactor": 0.7,
	"minbeamfrac": 0.3,
	"cutthreshold": 0.01
      }
    }
  },
  "Sgr_A_st_t_03_7M": {
    "tclean_cube_pars": {
      "spw24": {
        "imagename": "uid___A001_X15a0_X114.s38_0.Sgr_A_star_sci.spw24.cube.I.iter1.reclean",
        "threshold": "0.06Jy",
	"noisethreshold": 5.0,
	"lownoisethreshold": 2.0,
	"growiterations": 75,
	"sidelobethreshold": 1.25,
	"negativethreshold": 0.0,
	"smoothfactor": 0.5,
	"minbeamfrac": 0.1,
	"cutthreshold": 0.01
      }
    }
  }
}

**Comment by keflavich** [comment]
@pyhsiehATalma  Could you finish this PR?  There are just a few syntax changes needed, then we can merge it.

**Comment by pyhsiehATalma** [comment]
{
  "Sgr_A_st_c_03_7M": {
    "tclean_cont_pars": {
      "aggregate": {
        "imagename": "uid___A001_X15a0_Xae.s36_0.Sgr_A_star_sci.spw16_18_20_22_24_26.cont.I.iter1.reclean",
        "threshold": "0.0024Jy",
        "noisethreshold": 3.0,
        "lownoisethreshold": 1.75,
        "growiterations": 150
      }
    }
  },
  "Sgr_A_st_ar_03_7M": {
    "tclean_cube_pars": {
      "spw24": {
        "imagename": "uid___A001_X15a0_X1a4.s38_0.Sgr_A_star_sci.spw24.cube.I.iter1.reclean",
        "cyclefactor": 2.0
      },
      "spw26": {
        "imagename": "uid___A001_X15a0_X1a4.s38_0.Sgr_A_star_sci.spw26.cube.I.iter1.reclean",
        "cyclefactor": 2.0
      }
    }
  },
  "Sgr_A_st_ao_03_TM1": {
    "tclean_cube_pars": {
      "spw35": {
        "imagename": "uid___A001_X15a0_X190.s38_0.Sgr_A_star_sci.spw35.cube.I.iter1.reclean",
	"threshold": "0.01Jy",
	"noisethreshold": 3.25,
	"lownoisethreshold": 1.5,
	"growiterations": 75,
	"sidelobethreshold": 2.0,
	"negativethreshold": 10.0,
	"smoothfactor": 1.0,
	"minbeamfrac": 0.3,
	"cutthreshold": 0.01
      }
    }
  },
  "Sgr_A_st_ao_03_TM1": {
    "tclean_cube_pars": {
      "spw35": {
        "imagename": "uid___A001_X15a0_X190.s38_0.Sgr_A_star_sci.spw33.cube.I.iter1.reclean",
	"threshold": "0.013Jy",
	"noisethreshold": 3.25,
	"lownoisethreshold": 1.5,
	"growiterations": 75,
	"sidelobethreshold": 2.0,
	"negativethreshold": 15.0,
	"smoothfactor": 0.7,
	"minbeamfrac": 0.3,
	"cutthreshold": 0.01
      }
    }
  },
  "Sgr_A_st_t_03_7M": {
    "tclean_cube_pars": {
      "spw24": {
        "imagename": "uid___A001_X15a0_X114.s38_0.Sgr_A_star_sci.spw24.cube.I.iter1.reclean",
        "threshold": "0.06Jy",
	"noisethreshold": 5.0,
	"lownoisethreshold": 2.0,
	"growiterations": 75,
	"sidelobethreshold": 1.25,
	"negativethreshold": 0.0,
	"smoothfactor": 0.5,
	"minbeamfrac": 0.1,
	"cutthreshold": 0.01
      }
    }
  }
}

**Comment by pyhsiehATalma** [comment]
{
  "Sgr_A_st_c_03_7M": {
    "tclean_cont_pars": {
      "aggregate": {
        "imagename": "uid___A001_X15a0_Xae.s36_0.Sgr_A_star_sci.spw16_18_20_22_24_26.cont.I.iter1.reclean",
        "threshold": "0.0024Jy",
        "noisethreshold": 3.0,
        "lownoisethreshold": 1.75,
        "growiterations": 150
      }
    }
  },
  "Sgr_A_st_ar_03_7M": {
    "tclean_cube_pars": {
      "spw24": {
        "imagename": "uid___A001_X15a0_X1a4.s38_0.Sgr_A_star_sci.spw24.cube.I.iter1.reclean",
        "cyclefactor": 2.0
      },
      "spw26": {
        "imagename": "uid___A001_X15a0_X1a4.s38_0.Sgr_A_star_sci.spw26.cube.I.iter1.reclean",
        "cyclefactor": 2.0
      }
    }
  },
  "Sgr_A_st_ao_03_TM1": {
    "tclean_cube_pars": {
      "spw35": {
        "imagename": "uid___A001_X15a0_X190.s38_0.Sgr_A_star_sci.spw35.cube.I.iter1.reclean",
	"threshold": "0.01Jy",
	"noisethreshold": 3.25,
	"lownoisethreshold": 1.5,
	"growiterations": 75,
	"sidelobethreshold": 2.0,
	"negativethreshold": 10.0,
	"smoothfactor": 1.0,
	"minbeamfrac": 0.3,
	"cutthreshold": 0.01
      }
    }
  },
  "Sgr_A_st_ao_03_TM1": {
    "tclean_cube_pars": {
      "spw35": {
        "imagename": "uid___A001_X15a0_X190.s38_0.Sgr_A_star_sci.spw33.cube.I.iter1.reclean",
	"threshold": "0.013Jy",
	"noisethreshold": 3.25,
	"lownoisethreshold": 1.5,
	"growiterations": 75,
	"sidelobethreshold": 2.0,
	"negativethreshold": 15.0,
	"smoothfactor": 0.7,
	"minbeamfrac": 0.3,
	"cutthreshold": 0.01
      }
    }
  },
  "Sgr_A_st_t_03_7M": {
    "tclean_cube_pars": {
      "spw24": {
        "imagename": "uid___A001_X15a0_X114.s38_0.Sgr_A_star_sci.spw24.cube.I.iter1.reclean",
        "threshold": "0.06Jy",
	"noisethreshold": 5.0,
	"lownoisethreshold": 2.0,
	"growiterations": 75,
	"sidelobethreshold": 1.25,
	"negativethreshold": 0.0,
	"smoothfactor": 0.5,
	"minbeamfrac": 0.1,
	"cutthreshold": 0.01
      }
    }
  }
}

**Review by pyhsiehATalma — COMMENTED** [review]
{
  "Sgr_A_st_c_03_7M": {
    "tclean_cont_pars": {
      "aggregate": {
        "imagename": "uid___A001_X15a0_Xae.s36_0.Sgr_A_star_sci.spw16_18_20_22_24_26.cont.I.iter1.reclean",
        "threshold": "0.0024Jy",
        "noisethreshold": 3.0,
        "lownoisethreshold": 1.75,
        "growiterations": 150
      }
    }
  },
  "Sgr_A_st_ar_03_7M": {
    "tclean_cube_pars": {
      "spw24": {
        "imagename": "uid___A001_X15a0_X1a4.s38_0.Sgr_A_star_sci.spw24.cube.I.iter1.reclean",
        "cyclefactor": 2.0
      },
      "spw26": {
        "imagename": "uid___A001_X15a0_X1a4.s38_0.Sgr_A_star_sci.spw26.cube.I.iter1.reclean",
        "cyclefactor": 2.0
      }
    }
  },
  "Sgr_A_st_ao_03_TM1": {
    "tclean_cube_pars": {
      "spw35": {
        "imagename": "uid___A001_X15a0_X190.s38_0.Sgr_A_star_sci.spw35.cube.I.iter1.reclean",
	"threshold": "0.01Jy",
	"noisethreshold": 3.25,
	"lownoisethreshold": 1.5,
	"growiterations": 75,
	"sidelobethreshold": 2.0,
	"negativethreshold": 10.0,
	"smoothfactor": 1.0,
	"minbeamfrac": 0.3,
	"cutthreshold": 0.01
      }
    }
  },
  "Sgr_A_st_ao_03_TM1": {
    "tclean_cube_pars": {
      "spw35": {
        "imagename": "uid___A001_X15a0_X190.s38_0.Sgr_A_star_sci.spw33.cube.I.iter1.reclean",
	"threshold": "0.013Jy",
	"noisethreshold": 3.25,
	"lownoisethreshold": 1.5,
	"growiterations": 75,
	"sidelobethreshold": 2.0,
	"negativethreshold": 15.0,
	"smoothfactor": 0.7,
	"minbeamfrac": 0.3,
	"cutthreshold": 0.01
      }
    }
  },
  "Sgr_A_st_t_03_7M": {
    "tclean_cube_pars": {
      "spw24": {
        "imagename": "uid___A001_X15a0_X114.s38_0.Sgr_A_star_sci.spw24.cube.I.iter1.reclean",
        "threshold": "0.06Jy",
	"noisethreshold": 5.0,
	"lownoisethreshold": 2.0,
	"growiterations": 75,
	"sidelobethreshold": 1.25,
	"negativethreshold": 0.0,
	"smoothfactor": 0.5,
	"minbeamfrac": 0.1,
	"cutthreshold": 0.01
      }
    }
  }
}

**Comment by pyhsiehATalma** [comment]
Hi Adam,

It's a bit while ago and I am not sure I remembered how to make a change
request.
I attached the file of reclean here and it appears to be validated with jq
. ......

regards,
Pei-Ying

Adam Ginsburg ***@***.***> 於 2022年8月20日 週六 下午4:14寫道：

**Comment by keflavich** [comment]
@pyhsiehATalma OK, we can go over how to modify this pull request.  Adding text as a comment is not helpful, though, we need the change submitted properly.

**Comment by pyhsiehATalma** [comment]
yup, that would be helpful to remind me how to do that.

Adam Ginsburg ***@***.***> 於 2022年8月23日 週二 上午10:22寫道：

**Review by pyhsiehATalma — COMMENTED** [review]
{
  "Sgr_A_st_c_03_7M": {
    "tclean_cont_pars": {
      "aggregate": {
        "imagename": "uid___A001_X15a0_Xae.s36_0.Sgr_A_star_sci.spw16_18_20_22_24_26.cont.I.iter1.reclean",
        "threshold": "0.0024Jy",
        "noisethreshold": 3.0,
        "lownoisethreshold": 1.75,
        "growiterations": 150
      }
    }
  },
  "Sgr_A_st_ar_03_7M": {
    "tclean_cube_pars": {
      "spw24": {
        "imagename": "uid___A001_X15a0_X1a4.s38_0.Sgr_A_star_sci.spw24.cube.I.iter1.reclean",
        "cyclefactor": 2.0
      },
      "spw26": {
        "imagename": "uid___A001_X15a0_X1a4.s38_0.Sgr_A_star_sci.spw26.cube.I.iter1.reclean",
        "cyclefactor": 2.0
      }
    }
  },
  "Sgr_A_st_ao_03_TM1": {
    "tclean_cube_pars": {
      "spw35": {
        "imagename": "uid___A001_X15a0_X190.s38_0.Sgr_A_star_sci.spw35.cube.I.iter1.reclean",
	"threshold": "0.01Jy",
	"noisethreshold": 3.25,
	"lownoisethreshold": 1.5,
	"growiterations": 75,
	"sidelobethreshold": 2.0,
	"negativethreshold": 10.0,
	"smoothfactor": 1.0,
	"minbeamfrac": 0.3,
	"cutthreshold": 0.01
      }
    }
  },
  "Sgr_A_st_ao_03_TM1": {
    "tclean_cube_pars": {
      "spw35": {
        "imagename": "uid___A001_X15a0_X190.s38_0.Sgr_A_star_sci.spw33.cube.I.iter1.reclean",
	"threshold": "0.013Jy",
	"noisethreshold": 3.25,
	"lownoisethreshold": 1.5,
	"growiterations": 75,
	"sidelobethreshold": 2.0,
	"negativethreshold": 15.0,
	"smoothfactor": 0.7,
	"minbeamfrac": 0.3,
	"cutthreshold": 0.01
      }
    }
  },
  "Sgr_A_st_t_03_7M": {
    "tclean_cube_pars": {
      "spw24": {
        "imagename": "uid___A001_X15a0_X114.s38_0.Sgr_A_star_sci.spw24.cube.I.iter1.reclean",
        "threshold": "0.06Jy",
	"noisethreshold": 5.0,
	"lownoisethreshold": 2.0,
	"growiterations": 75,
	"sidelobethreshold": 1.25,
	"negativethreshold": 0.0,
	"smoothfactor": 0.5,
	"minbeamfrac": 0.1,
	"cutthreshold": 0.01
      }
    }
  }
}

File: pipeline_scripts/override_tclean_commands.json (hunk 1 above)
Comment (keflavich) [reply pipeline_scripts/override_tclean_commands.json:40]: the current version of this pull request is syntactically incorrect; a comma is needed here in order for this code to be valid JSON

**Comment by pyhsiehATalma** [comment]
Hi Adam,

I feel so sorry to take this long. This request appears to be a bit messy
now.
I was working on reclean of 4 EBs, but the original pull request has only
two EBs (https://github.com/ACES-CMZ/reduction_ACES/pull/94)
It's unclear to me how to update the file or just create a new pull request.

Sgr_A_st_ar_03_7M spw 24
Sgr_A_st_ao_03_TM1 spw 35
Sgr_A_st_ao_03_TM1 spw 33
Sgr_A_st_t_03_7M spw 24

https://github.com/ACES-CMZ/reduction_ACES/blob/main/aces/pipeline_scripts/override_tclean_commands.json

Pei-Ying Hsieh ***@***.***> 於 2022年8月23日 週二 上午10:31寫道：

**Comment by keflavich** [comment]
Do you have a local clone of this repository?  Or are you only editing things via the github interface?

**Comment by pyhsiehATalma** [comment]
I don't have a local clone of this repository and just edit things via
the github interface.

Adam Ginsburg ***@***.***> 於 2022年8月24日 週三 下午3:04寫道：

**Comment by keflavich** [comment]
OK, then what you need to do is to edit the files in this pull request, _not_ the comments.

You can do this by going to the "Files Changed" tab, then on the top-right there is a `...` menu, under which you can click `Edit`

**Comment by pyhsiehATalma** [comment]
Thanks! That tab is something I was looking for. I edited the file and
which option should I go for?

Commit directly to the patch-1 branch.
Create a new branch for this commit and start a pull request. Learn
more about pull requests.


Adam Ginsburg ***@***.***> 於 2022年8月24日 週三 下午3:08寫道：

**Comment by pyhsiehATalma** [comment]
I updated the file and saved with "Commit directly to the patch-1 branch."
jq . override_tclean_commands.json appears to work, let me know if I need to do anything more.

**Comment by keflavich** [comment]
Looks good now.  Because this got out of date, we have some conflicts, so I'll try to resolve those.

**Pull request closed without merging**